### DISCUSSION TITLE
Making images respond to Light and Dark theme change.

### DISCUSSION THREAD

**Discussion by leigh-pointer** [twbs/bootstrap#38776] [Q&A]
I'm currently trying to workout the best way to handle images when the theme changes from light to dark using bs-5.3.  Any advice or ideas would be welcome

**Comment by ffoodd** [comment]
What would you want to do, exactly? Switch source? Apply filtres? Anything else?

**Reply by leigh-pointer** [reply]
Some images have light backgrounds and need to look good in the dark theme and vise versa, so switching source and/or applying filters.

**Reply by ffoodd** [reply]
If you use [the media query color mode type](https://getbootstrap.com/docs/5.3/customize/color-modes/#building-with-sass), you may use the `<picture>` element to provide different sources depending on media queries, eg.:

```html
<picture>
    <source srcset="mojave-night.jpg" media="(prefers-color-scheme: dark)">
    <img src="mojave-day.jpg">
</picture>
```

See [the WebKit blog post for dark mode announcement back in 2019](https://webkit.org/blog/8840/dark-mode-support-in-webkit/).

If you use themes based on `data-bs-theme`, I’m afraid there's no really good solution:
1. either use JavaScript to check for `data-bs-theme` value and switch source accordingly,
2. or add every sources in your markup from the start, and use CSS to hide them (eg. `[data-bs-theme="dark"] img[src*="light"] { display: none; }`).

As of now, there's no straightforward way to accomplish this with Bootstrap only.

**Comment by patkon** [comment]
For background images you can try something like that (untested)
```
.yourClass {
background-image: url("light_image.png");
}

[data-bs-theme=dark] .yourClass {
background-image: url("dark_image.png");
}
```
Making all images dependent on the theme feels resource-hungry to me. You would have to check the current theme with javascript and then decide which image to display. But that also means you have to load all images twice (dark/light).

**Reply by patkon** [reply]
Here's a Javascript Solution
https://github.com/orgs/twbs/discussions/38793

**Comment by W3AXL** [comment]
Another option that worked for me was making utility classes to hide items when the opposite theme is selected:

```css
/* Hide light-mode things when in dark mode */
[data-bs-theme=dark] .display-light {
    display: none;
}
/* Hide dark-mode things when in light mode */
[data-bs-theme=light] .display-dark {
    display: none;
}
```

Then I load two images and apply the appropriate classes:
```html
<img class="display-dark" src="logo-dark.png">
<img class="display-light" src="logo-light.png">
```

Maybe not the most resource-efficient, but simple if you're only using it for select images.
